cognite/client/data_classes/datapoints.py

```diff
@@ -271,19 +333,69 @@ def __str__(self):
             i["datapoints"] = utils._time.convert_time_attributes_to_datetime(i["datapoints"])
         return json.dumps(item, default=lambda x: x.__dict__, indent=4)
 
-    def to_pandas(self, column_names="externalId") -> "pandas.DataFrame":
+    def to_pandas(
+        self, column_names="externalId", include_aggregate_name=True, complete=None, complete_granularity=None
+    ) -> "pandas.DataFrame":
         """Convert the datapoints list into a pandas DataFrame.
 
         Args:
             column_names (str): Which field to use as column header. Defaults to "externalId", can also be "id".
+            include_aggregate_name (bool): Include |aggregate in the column name
+            complete (bool): Post-processing of the dataframe.
+                Pass 'fill' to insert missing entries into the index, and complete data where possible (supports interpolation, stepInterpolation, count, sum, totalVariation).
+                Pass 'fill,dropna' to additionally drop rows in which any aggregate for any time series has missing values (typically rows at the start and end for interpolation aggregates). Only supported if all aggregates can be completed.
+            complete_granularity (str): When complete=True, what granularity should the index be given. Defaults to the minimum difference between timestamps.
+
         Returns:
             pandas.DataFrame: The datapoints list as a pandas DataFrame.
         """
         pd = utils._auxiliary.local_import("pandas")
-        dfs = [df.to_pandas(column_names=column_names) for df in self.data]
-        if dfs:
-            return pd.concat(dfs, axis="columns")
-        return pd.DataFrame()
+
+        complete = [s.strip() for s in (complete or "").split(",")]
+        if set(complete) - set(["fill", "dropna", ""]):
+            raise ValueError("complete should be 'fill', 'fill,dropna' or Falsy")
+        fill_flag = "fill" in complete
+        is_step_dict = {dp.id: False for dp in self.data}
+        complete_start = None
+        complete_end = None
+        if "fill" in complete:
+            interp_ids = [dp.id for dp in self.data if dp.interpolation is not None]
+            starts = [dp.timestamp[0] for dp in self.data if dp.timestamp]
+            ends = [dp.timestamp[-1] for dp in self.data if dp.timestamp]
+            complete_start = min(starts) if starts else None
+            complete_end = max(ends) if starts else None
+            if interp_ids:  # get isStep flag and pass it to completion to handle interpolation correctly
+                ts_meta = self._cognite_client.time_series.retrieve_multiple(ids=interp_ids)
+                is_step_dict.update({ts.id: bool(ts.is_step) for ts in ts_meta})
+        dfs = [
+            df.to_pandas(
+                column_names=column_names,
+                include_aggregate_name=include_aggregate_name,
+                complete=fill_flag,
```

Comment:
won't this just always be True?

Reply:
fill_flag = "fill" in complete
(maybe clearer in the refactored version)